src/Serializers/FingerprintSerializer.php

```diff
@@ -47,25 +53,22 @@ public function addIdToSerialization( Entity $entity, array &$serialization ) {
 	}
 
 	public function addLabelsToSerialization( Entity $entity, array &$serialization ) {
-		$labels = $entity->getLabels();
+		$labels = $entity->getFingerprint()->getLabels();
```

Comment:
Why do you need "the actual object" and what does "actual" mean in the first place?